### PR TITLE
Re-introduce Tiptap editor with sanitization fixes

### PR DESCRIPTION
**Pull request by Rachit7168**
This PR re-introduces the Tiptap editor from #4147 which was reverted in #4520. It also includes the HTML sanitization and sourcemap fixes from #4511.

https://github.com/user-attachments/assets/22f1e52b-c579-41ba-ae84-cf52a6583b9f

## Summary by Sourcery

Reintroduce a JS-based Tiptap editor for rich text and email bodies, backed by new server-side HTML sanitization and preview mechanisms, and integrate it into existing email workflows and build infrastructure.

New Features:
- Add Tiptap-based rich text and email body editor widgets, including placeholder insertion and preview support.
- Introduce server-side HTML sanitization profiles for rich text and email editor output.

Bug Fixes:
- Ensure email HTML sanitization preserves mailto/tel links and avoids injecting rel attributes that can confuse email clients.
- Fix rich text link rendering to avoid altering existing link labels and adjust absolute link behaviour.

Enhancements:
- Refine snippet rendering to unwrap single outer paragraph tags for cleaner inline display.
- Update email compilation logic to distinguish between legacy Markdown/plain text and already-HTML Tiptap content.

Build:
- Add Vite-based build pipeline for the new editor webapp and integrate it into the Makefile.
- Register the editor JS/CSS bundles with the Django templates via a lightweight loader script.

Tests:
- Add tests for rich text linkification behaviour and email body compilation logic.
- Add tests covering HTML sanitizers for rich text and email profiles, including edge cases for schemes and attributes.
- Add tests for i18n email body form field sanitization behaviour.

Chores:
- Wire up new email editor fields in sendmail and control order forms, including placeholder lists and preview endpoints.
- Introduce a server-side email preview AJAX endpoint used by the editor toolbar.

### CODE REVIEW

**Review by ArnavBallinCode — CHANGES_REQUESTED** [review]
Remove QR placeholder commits of @Saksham-Sirohi , he has raised a PR for that separately.

**Comment by Rachit7168** [comment]
@ArnavBallinCode , 
Updated video and ur comment as well

**Review by ArnavBallinCode — APPROVED** [review]
Ready to test on dev

**Review by Saksham-Sirohi — APPROVED** [review]

File: app/eventyay/webapp/editor/vite.config.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import { defineConfig } from 'vite'
```
Comment (Saksham-Sirohi) [inline app/eventyay/webapp/editor/vite.config.js]: What is the need to create the whole editor as a new web app

**Review by Saksham-Sirohi — CHANGES_REQUESTED** [review]

File: app/eventyay/webapp/editor/vite.config.js [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+import { resolve } from 'node:path'
+import { fileURLToPath } from 'node:url'
+
+const __dirname = fileURLToPath(new URL('.', import.meta.url))
+
```
Comment (ArnavBallinCode) [inline app/eventyay/webapp/editor/vite.config.js]: This was the entire reason the first PR was reverted, same again?

**Review by ArnavBallinCode — CHANGES_REQUESTED** [review]

**Comment by Rachit7168** [comment]
Needs discussion regarding approved not for review

**Comment by Rachit7168** [comment]
https://github.com/user-attachments/assets/fa6a95fd-ed24-4645-b18e-9d5ce2ed6e4f

**Review by Saksham-Sirohi — APPROVED** [review]
LGTM using vendor files is a nice addition

**Review by ArnavBallinCode — APPROVED** [review]
LGTM. Lets test on dev
